sdk/core/switcher/entry.S

```diff
@@ -779,3 +779,77 @@ exception_entry_asm:
 #endif
 #endif // CONFIG_NO_SWITCHER_SAFETY
 	cret
+
+// Returns whether the trusted stack has space for N more calls.
+	.section .text, "ax", @progbits
+	.p2align 2
+	.type __Z23trusted_stack_has_spacei,@function
+__Z23trusted_stack_has_spacei:
+	li                 a2, TrustedStackFrame_size
+	mul                a2, a0, a2
+	cspecialr          ca0, mtdc
+	clhu               a1, TrustedStack_offset_frameoffset(ca0)
+	cgetlen            a0, ca0
+	sub                a0, a0, a1
+	sltu               a0, a2, a0
+	cret
+
+// Returns the size of the trusted stack.
+	.section .text, "ax", @progbits
+	.p2align 2
+	.type __Z18trusted_stack_sizev,@function
+__Z18trusted_stack_sizev:
+	cspecialr          ca0, mtdc
+	cgetlen            a0, ca0
+	j                  size_to_trusted_stack_frames
```

Comment:
Not that I object, but this jumping over another exported symbol seems like the kind of thing that's going to confuse a debugger?

Reply:
No, debuggers know how to handle tail merging.  Mind you, a debugger looking at assembly will not give you anything very interesting.